presto-main/src/main/java/io/prestosql/eventlistener/EventListenerManager.java

```diff
@@ -79,8 +80,10 @@ protected void setConfiguredEventListener(String name, Map<String, String> prope
         EventListenerFactory eventListenerFactory = eventListenerFactories.get(name);
         checkState(eventListenerFactory != null, "Event listener %s is not registered", name);
 
-        EventListener eventListener = eventListenerFactory.create(ImmutableMap.copyOf(properties));
-        this.configuredEventListener.set(Optional.of(eventListener));
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(eventListenerFactory.getClass().getClassLoader())) {
```

Comment:
This should be done inside the plugin. It's generally needed for certain libraries that don't do "the right thing", so the engine should not be involved. Why can't we do it like every other ClassLoaderSafeXXX variant and let plugins do the wrapping if necessary.

Reply:
Oh, just saw the comment in the other PR. Let me think about it more.